ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -82,19 +87,26 @@ public void setPrivateTransactionProcessor(
 
   @Override
   public Gas gasRequirement(final BytesValue input) {
-    return Gas.of(40_000L); // Not sure
+    return Gas.of(0L);
   }
 
   @Override
   public BytesValue compute(final BytesValue input, final MessageFrame messageFrame) {
+    final ProcessableBlockHeader currentBlockHeader = messageFrame.getBlockHeader();
+    if (!BlockHeader.class.isAssignableFrom(currentBlockHeader.getClass())) {
+      LOG.info("Privacy works with BlockHeader only!.");
```

Comment:
This isn't expected in a "normal" execution, is it? I would change it from INFO to WARN level.

Reply:
If this scenario can never happen, I'd even go further and crash the node (too harsh?)

Reply:
Agreed, if it can never happen we should crash the node and understand how/why it did crash!